source/extensions/filters/udp/udp_proxy/udp_proxy_filter.h

```diff
@@ -169,6 +186,14 @@ class UdpProxyFilter : public Network::UdpListenerReadFilter,
   private:
     void onIdleTimer();
     void onReadReady();
+    void fillStreamInfo();
+
+    UdpProxyDownstreamStats generateStats(Stats::Scope& scope) {
+      const auto final_prefix =
+          absl::StrCat("udp.", cluster_.filter_.config_->randomGenerator().uuid());
```

Comment:
This is not going to work. It will lead to stats explosion and break both Envoy and your monitoring system pretty badly. I would just use a normal struct of non-atomic values, since there is no sharing between threads. You don't need real stats for this.

Reply:
> This is not going to work. It will lead to stats explosion and break both Envoy and your monitoring system pretty badly. I would just use a normal struct of non-atomic values, since there is no sharing between threads. You don't need real stats for this.

OK, I have reimplemented using struct.